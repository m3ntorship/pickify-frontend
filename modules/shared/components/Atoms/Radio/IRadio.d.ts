declare namespace IRadio {
  export interface IProps {
    size: string;
    disabled: boolean;
  }
}
export { IRadio };
