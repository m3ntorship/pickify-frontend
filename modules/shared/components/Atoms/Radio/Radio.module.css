.radio_transparency{
    @apply absolute opacity-0 cursor-pointer;
}
.radio_background{
    @apply bg-white rounded-full;
}
.size_small{
    @apply w-4 h-4;
}
.size_default{
    @apply w-6 h-6;
}
.checked_small{
    @apply border-primary border-4 hover:border-primary-hover focus:ring-2 focus:ring-offset-primary-shd6 focus:outline-none;
}
.checked_default{
    @apply border-primary border-8 hover:border-primary-hover focus:ring-2 focus:ring-offset-primary-shd6 focus:outline-none;
}
.non_checked{
    @apply border border-grey-shd4 hover:border-grey-shd1 focus:ring-1 focus:ring-offset-grey-shd2 focus:outline-none;
}
.disabled{
    @apply pointer-events-none opacity-25;
}

