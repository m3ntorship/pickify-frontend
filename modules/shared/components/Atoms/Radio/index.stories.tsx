import React from 'react';
import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import Radio from '.';
import type { IRadio } from './IRadio';

export default {
  component: Radio,
  title: 'Atoms/Radio',
  argTypes: {
    size: {
      options: ['small', 'default'],
      control: { type: 'select' },
    },
    label: {
      control: {
        type: 'text',
      },
    },
    defaultChecked: {
      control: {
        type: 'boolean',
      },
    },
    disabled: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const Template: Story<IRadio.IProps> = (args): ReactElement => (
  <Radio {...args} />
);

export const RadioTS = Template.bind({});
RadioTS.args = {
  size: 'default',
  label: 'test',
  defaultChecked: false,
  disabled: false,
};
