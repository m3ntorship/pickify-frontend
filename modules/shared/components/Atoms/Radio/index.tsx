import type { ReactElement, FC } from 'react';
import React, { useState } from 'react';
import classnames from 'classnames';
import styles from './Radio.module.css';
import type { IRadio } from './IRadio';

const Radio: FC<IRadio.IProps> = ({
  size = 'default',
  disabled = false,
}): ReactElement => {
  const [checked, setChecked] = useState(false);
  const radioClasses = classnames(styles.radio_background, {
    [styles.size_small]: size === 'small',
    [styles.size_default]: size === 'default',
    [styles.checked_small]: checked && size === 'small',
    [styles.checked_default]: checked && size === 'default',
    [styles.non_checked]: !checked,
    [styles.disabled]: disabled,
  });
  const radioInputClasses = classnames(styles.radio_transparency, {
    [styles.size_small]: size === 'small',
    [styles.size_default]: size === 'default',
  });

  return (
    <div className="relative">
      <input
        type="radio"
        className={radioInputClasses}
        data-testid="Radio"
        disabled={disabled}
        onClick={(): void => {
          setChecked(!checked);
        }}
      />
      <div className={radioClasses} />
    </div>
  );
};

export default Radio;
