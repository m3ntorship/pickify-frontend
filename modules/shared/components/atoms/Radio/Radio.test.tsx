import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Radio from '.';

describe('Testing Radio with snapshot', () => {
  it('Testing Radio with required props', () => {
    const tree = renderer
      .create(
        <Radio
          size="default"
          name="laptops"
          id="dell"
          value="dell"
          defaultChecked={false}
          onChange={(): string => {
            return 'test';
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Testing Radio with required and optional props', () => {
    const tree = renderer
      .create(
        <Radio
          size="small"
          name="laptops"
          id="dell"
          value="dell"
          defaultChecked={false}
          onChange={(): string => {
            return 'test';
          }}
          disabled
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing Radio with RTL', () => {
  it('Expect Radio is not to be checked', () => {
    const { getByTestId } = render(
      <Radio
        size="default"
        name="laptops"
        id="dell"
        value="dell"
        defaultChecked={false}
        onChange={(): string => {
          return 'test';
        }}
      />,
    );
    expect(getByTestId('Radio')).not.toBeChecked();
  });
  it('Expect Radio is checked when defaultChecked is true', () => {
    const { getByTestId } = render(
      <Radio
        size="default"
        name="laptops"
        id="dell"
        value="dell"
        defaultChecked
        onChange={(): string => {
          return 'test';
        }}
      />,
    );
    expect(getByTestId('Radio')).toBeChecked();
  });
  it('Expect Radio is checked after clicking', () => {
    const { getByTestId } = render(
      <Radio
        size="default"
        name="laptops"
        id="dell"
        value="dell"
        defaultChecked={false}
        onChange={(): string => {
          return 'test';
        }}
      />,
    );
    userEvent.click(getByTestId('Radio'));
    expect(getByTestId('Radio')).toBeChecked();
  });
  it('Expect Radio is disabled', () => {
    const { getByTestId } = render(
      <Radio
        size="default"
        name="laptops"
        id="dell"
        value="dell"
        defaultChecked={false}
        onChange={(): string => {
          return 'test';
        }}
        disabled
      />,
    );
    expect(getByTestId('Radio')).toBeDisabled();
  });
  it('Expect onChange is called once', () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <Radio
        size="default"
        name="laptops"
        id="dell"
        value="dell"
        defaultChecked={false}
        onChange={handleChange}
      />,
    );
    userEvent.click(getByTestId('Radio'));
    expect(handleChange).toHaveBeenCalled();
  });
});
