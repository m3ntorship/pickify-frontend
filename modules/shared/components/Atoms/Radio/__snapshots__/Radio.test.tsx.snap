// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Radio with snapshot Testing Radio with required and optional props 1`] = `
<div>
  <input
    className=""
    data-testid="Radio"
    defaultChecked={false}
    disabled={true}
    id="dell"
    name="laptops"
    onChange={[Function]}
    type="radio"
    value="dell"
  />
  <div
    className=""
  />
</div>
`;

exports[`Testing Radio with snapshot Testing Radio with required props 1`] = `
<div>
  <input
    className="undefined"
    data-testid="Radio"
    defaultChecked={false}
    disabled={false}
    id="dell"
    name="laptops"
    onChange={[Function]}
    type="radio"
    value="dell"
  />
  <div
    className=""
  />
</div>
`;
